src/Resources/skeleton/forgottenPassword/PasswordResetToken.tpl.php

```diff
@@ -0,0 +1,87 @@
+<?= "<?php\n" ?>
+
+namespace <?= $namespace ?>;
+
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Entity()
+ */
+class <?= $class_name ?>
+
+{
+    const TTL_HOURS = 24;
+    const SELECTOR_LENGTH = 20; // in chars
+
+    /**
+     * @ORM\Id()
+     * @ORM\GeneratedValue()
+     * @ORM\Column(type="integer")
+     */
+    private $id;
+
+    /**
+     * @ORM\Column(type="string")
+     */
+    private $selector;
+
+    /**
+     * @ORM\Column(type="string")
+     */
+    private $token;
+
+    /**
+     * @ORM\Column(type="datetime")
+     */
+    private $requestedAt;
+
+    /**
+     * @ORM\ManyToOne(targetEntity="<?= $user_full_class_name ?>")
+     */
+    private $user;
+
+    private $plainToken;
+
+    public function __construct(User $user)
+    {
+        $this->requestedAt = new \DateTimeImmutable('now');
+        $this->selector = strtr(base64_encode(random_bytes(self::SELECTOR_LENGTH * 3 / 4)), '+/', '-_');
+        $this->plainToken = strtr(base64_encode(random_bytes(18)), '+/', '-_');
+        $this->token = password_hash($this->plainToken, PASSWORD_BCRYPT);
+        $this->user = $user;
+    }
+
+    public function getId()
+    {
+        return $this->id;
+    }
+
+    public function getAsString(): string
+    {
+        if (!$this->selector || !$this->plainToken) {
+            throw new \Exception('You can get PasswordResetToken as a string only immediately after creation.');
+        }
+
+        return $this->selector.$this->plainToken;
+    }
+
+    public function getUser(): <?= $user_class_name ?>
+
+    {
+        return $this->user;
+    }
+
+    public function isTokenEquals(string $token): bool
+    {
+        return password_verify($token, $this->token);
+    }
```

Comment:
And I think I'd like to use `NativePasswordEncoder` here too. Well, not here of course :). But probably do this check in the controller.